packages/core/src/html/SiteLinkManager.ts

```diff
@@ -22,11 +27,12 @@ class SiteLinkManager {
    * Adds a resourcePath and cwf to the intralinkCollection,
    * ensuring each pair of (resourcePath, cwf) appears only once
    */
-  _addToCollection(resourcePath, cwf) {
+  _addToCollection(resourcePath: string, cwf: string) {
     if (!this.intralinkCollection.has(cwf)) {
       this.intralinkCollection.set(cwf, new Set());
     }
-    this.intralinkCollection.get(cwf).add(resourcePath);
+    // We have checked and set cwf in intralinkCollection above
+    this.intralinkCollection.get(cwf)!.add(resourcePath);
```

Comment:
The use of `!.` is to get rid of the warnings?

Reply:
`!` asserts that the stuff before it (`this.intralinkCollection.get(cwf)`) is not null in order to remove the compiler warnings. I added a comment to document why it is safe to do so (because we have checked using `has` and done a `set` if it doesn't exist)!